### DISCUSSION TITLE
How should the logger process its arguments

### DISCUSSION THREAD

**Discussion by stefaanMLB** [sequelize/umzug#326] [Q&A]
I need to log the umzug messages to Winston.
Setting logger=console.log works fine.  Assigning a Winston logger doesn't work.
For now I'm using `logging: (...args: any[]) =>log.info(args[0]);` which works but doesn't seem right.
Could you document exactly how the logging function should process the `...args` arguments ?

**Comment by mmkal** [comment]
Right now, only plain strings are logged, so your solution will work. But it may make sense to change the log function signature to `(message: string) => void` to make this clearer.

In https://github.com/sequelize/umzug/pull/325 `logging` is replaced with a `logger` (with info, warn and debug methods). It should be possible to test that it accepts a few different kinds of loggers, including winston.

**Comment by mmkal** [comment]
@stefaanMLB having changed the argument type to `string` in the beta release, I'm now thinking it'd make sense to broaden it to `any` or `unknown` before v3 stable is published. That will allow us to produce structured, json-able logs suitable for production log processors and query tools. It won't be `any[]` though, we can keep it limited to one arg.

**Comment by stefaanMLB** [comment]
I love json-able logs.
You have my blessing
